shared/src/main/scala/org/atnos/eff/Member.scala

```diff
@@ -371,23 +293,23 @@ trait MemberLower6 extends MemberLower7 {
     type Out = Fx3[T, L, R]
 
     def inject[V](m: M[V]): Union[FxAppend[Fx1[T], Fx3[L, M, R]], V] =
-      UnionAppendR[Fx1[T], Fx3[L, M, R], V](Union3M(m))
+      Union.appendR(Union.threeM(m))
 
-    def accept[V](union: Union[Fx3[T, L, R], V]): Union[FxAppend[Fx1[T], Fx3[L, M, R]], V] =
-      union match {
-        case Union3L(t) => UnionAppendL(Union1(t))
-        case Union3M(l) => UnionAppendR(Union3L(l))
-        case Union3R(r) => UnionAppendR(Union3R(r))
+    def accept[V](union: Union[Fx3[T, L, R], V]): Union[FxAppend[Fx1[T], Fx3[L, M, R]], V] = union.tagged match {
+      case tagged@UnionTagged(_, index) => (index: @switch) match {
+        case 1 => UnionAppendR(tagged.forget)
+        case _ => UnionAppendL(tagged.xor())
```

Comment:
I am not sure the tests in `MemberImplicitsSpec` will catch this issue because the commented out tests are more about implicits inference. In order to exercice this `accept` capability you would have to use something like `transform` with a specific stack I think.